content/blog/gatsby-blog/index.md

```diff
@@ -0,0 +1,323 @@
+
+But here is the gist:
+
+- **for apex domain**: create an 4 `A` records, each with one of these IPs
+```bash
+185.199.108.153
+185.199.109.153
+185.199.110.153
+185.199.111.153
+```
+- **for `www.`**: create a `CNAME` record with a key of `www` and a value of `username.github.io`
+
+You may have both without a problem
+
+
+
+### 2 Setup you CNAME in github
+
+If you are using `npm run deploy` to deploy you app then go to your repo's settings,
+scroll to the bottom and add the custom domain in the **Custom Domain** input.
+You should add your naked or apex domain, without the `www`.
+
+If you are using github actions (more on this later) then you need to add
+this exact file to your source code in the `develop` branch,
+it should be named exactly `CNAME` and it should contain your domain as the only line of text.
+
+CNAME
+```bash
+myblog.com
+```
+
+
+Depending on your domain provider these changes in the DNS records might take
+from a couple of minutes to a couple of days (in our case it was pretty fast with godaddy).
+
+Checkout the `dig` linux/unix command to diagnose DNS updates (probably will do a blog post about this)
+
+
+
+
+## Deploy automatically with Github Actions (CD)
+
+[Github Actions](https://help.github.com/en/actions/automating-your-workflow-with-github-actions) is an automation
+service provided by Github with free tiers and quick access if you host your code in github.
+You may use other automation tools for this but the steps will definitively vary greatly.
+
+A bit of a glossary:
+- actions: a list of steps to perform a reusable task, it accepts parameters (yes, the service is called "Github actions" and one of the unit of abstractions inside the service are also called "actions").
+- workflow: an automation that will executed by github when some event that you define happens (i.e. push to a branch). It will run a number of steps that might be actions or simple bash commands
+
+### 1 Create a workflow
+
+Create the workflow file in your `develop` branch at `.github/workflows/deploy.yml`
+You can call it whatever you like as long as it is inside that directory.
+
+Add this to the file contents
+
+```yml
+name: Gatsby Publish
+
+on:
+  push:
+    branches:
+      - develop
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - uses: enriikke/gatsby-gh-pages-action@v2
+      with:
+        access-token: ${{ secrets.ACCESS_TOKEN }}
+        deploy-branch: master
+```
+
+
+You may create this via the github actions UI in your repo's github page.
+
+
+Notice we are using two actions, one is a standard action provided
+by github to download the repo's code in the
+automation runner (think of it as a virtual machine) and
+the other one [`gatsby-gh-pages`](https://github.com/enriikke/gatsby-gh-pages-action) is a third party provided one.
+
+I inspected the code of the latter and it seemed ok but I always recommend checking third party
```

Comment:
nit: ok = OK
https://www.writing-skills.com/ok-ok-okay-how-do-you-write-ok
https://www.lifehacker.com.au/2013/08/its-not-ok-to-write-ok/